drivers/led/led_pwm.c

```diff
@@ -20,7 +20,7 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(led_pwm, CONFIG_LED_LOG_LEVEL);
 
-#define DEV_CFG(dev)	((const struct led_pwm_config *) ((dev)->config))
+#define DEV_CFG(dev)	((struct led_pwm_config *) ((dev)->config))
```

Comment:
This should not be changed. The structure is read-only: https://github.com/zephyrproject-rtos/zephyr/blob/376e8b1366a6303771f9a70f002e9d4d83cbf750/drivers/led/led_pwm.c#L219-L222 so the compiler is free to locate it in flash.
You should instead use the `dev->data` structure and store the PM state there (the data structure was present in this driver but it has been [removed some time ago](https://github.com/zephyrproject-rtos/zephyr/commit/f2ebf488d7b197e09e50852f95e93ad5d24725fc#diff-40d01f15a18a2653eb2c631b832562793f9576f19878f6a67f3ee4e7bf451242L24) when it became unneeded; it can be restored now).

Reply:
Is the increased complication worth it? `led_pwm_##id` array will stay on flash anyway so the saving is only in keeping on flash one pointer and one int per device.

Reply:
The one thing that must be fixed for sure is making the change match with structure definition as it may lead to fault when compiler indeed move the structure to flash.

Reply:
> You should instead use the `dev->data` structure and store the PM state there (the data structure was present in this driver but it has been [removed some time ago](https://github.com/zephyrproject-rtos/zephyr/commit/f2ebf488d7b197e09e50852f95e93ad5d24725fc#diff-40d01f15a18a2653eb2c631b832562793f9576f19878f6a67f3ee4e7bf451242L24) when it became unneeded; it can be restored now).

Good catch! @pdunaj Please use dev_data for storing the PM state. It should belong there.

Reply:
will push change in a minute :)

Reply:
done